### ISSUE TITLE
[Wave 8 Ideal Nav] Refreshing page on a Workspace item page causes double navigation

### ISSUE THREAD

**Issue by hayata-suenaga** [Expensify/App#35614]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**

## Action Performed:
1. Shrink your web browser
2. 🔧 Workspace Settings > Reimbursements > refresh the page > Press your browser's back button

## Expected Result:
The back button should take you to the Workspace Settings page

## Actual Result:
After the back button is pressed, you're still on the Reimbursements page. You need to click the back button once more to go back to the Settings page.

## Workaround:
N/A

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
 
video ->  https://github.com/Expensify/App/pull/33280#issuecomment-1911319957

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by greg-schroeder** [comment]
@ntdiary do you mind reviewing the proposal above? thanks!

**Comment by bernhardoj** [comment]
My proposal [here](https://github.com/Expensify/App/issues/35692) can solve this too.

**Comment by hayata-suenaga** [comment]
@bernhardoj could you double post the solution here if you can? and if you can explain how your solution solves both issues at the same time, that would be super helpful 🙇

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The app navigates to the reimbursement page twice when refreshing the web, so we need to press back twice to close it.

### What is the root cause of that problem?
When we open a URL on the web, it will open the corresponding page and at the same time call Report.openReportFromDeeplink which will call an extra navigation call with a PUSH type.
https://github.com/Expensify/App/blob/3ab4e6e12d9ff38e69493321f0d31f032481b373/src/libs/actions/Report.ts#L2164

We already have a logic to not apply the PUSH action if the route destination is the active route.
https://github.com/Expensify/App/blob/3ab4e6e12d9ff38e69493321f0d31f032481b373/src/libs/Navigation/linkTo.ts#L246-L248

But after the ideal nav PR, we modify this logic to change the action type to PUSH if the destination is a central pane navigator (which is true for reimbursement page) but the current central pane screen is not a report screen, so we always push another reimbursement page to the stack (or any other central pane route)
https://github.com/Expensify/App/blob/3ab4e6e12d9ff38e69493321f0d31f032481b373/src/libs/Navigation/linkTo.ts#L162-L179

### What changes do you think we should make in order to solve the problem?

We should check whether the route destination is the same as the current central pane route instead of checking whether it's a report screen or not. If it's the same, then don't push it.
```
topmostCentralPaneRoute.name !== action.payload.params.screen
```

This will solve https://github.com/Expensify/App/issues/35692 too with the same root cause explained [there](https://github.com/Expensify/App/issues/35692#issuecomment-1925092468)

**Comment by hayata-suenaga** [comment]
@ntdiary could you review the proposals when you have time 🙇 ?

**Comment by hayata-suenaga** [comment]
I don't think there is a mention about the condition you mentioned. I'll get @adamgrzybowski check on this condition statement.

**Comment by adamgrzybowski** [comment]
@ntdiary  I think this was just overlooked in the process. You are right we should to check if the destination is the same as current topmost central pane.

We still need to have a separate check for Report route though. We can push multiple different report screens and in that case we need to compare reportID.

**Comment by hayata-suenaga** [comment]
makes sense. @ntdiary if you still think @bernhardoj's proposal is the best, please post the approval comment 🙇

**Comment by bernhardoj** [comment]
PR is ready

cc: @ntdiary

**Comment by greg-schroeder** [comment]
PR Merged and deployed to staging. Awaiting deploy to prod

**Comment by greg-schroeder** [comment]
Paid, @ntdiary can you complete the checklist? Thanks!

**Comment by greg-schroeder** [comment]
ahh, okay, works for me!

**Comment by greg-schroeder** [comment]
In that case the payment summary would be:

@ntdiary as C+ $500 - manual request
@bernhardoj as C $500

**Comment by greg-schroeder** [comment]
Given you'll be making a ND manual request and regression test is filed, closing! If you end up needing to be paid through Upwork, please reopen this.
